src/main/java/org/broadinstitute/hellbender/engine/AlignmentContext.java

```diff
@@ -140,10 +146,7 @@ public AlignmentContext stratify(final ReadOrientation type) {
             if(sample != null) {
```

Comment:
note that the old code and the new code it still bogus - if `sample == null` then the the line up (`sample.equals`) blows up and so this check is useless. I'm fixing it and adding tests.